### ISSUE TITLE
Plasma 5 support

### ISSUE THREAD

**Issue by christophschw** [carlin-q-scott/browser-media-keys#66]
Hey,

thanks for this useful plugin! It works very well in Fedora 23 with Gnome, but my primary DE is Plasma 5 - so most of the time I can't use it.

So is it possible to start the processes the extension need even in another DE?

Would be great to have it working. Thanks,

Christoph

**Comment by carlin-q-scott** [comment]
Hi Christoph,

The add-on uses DBus Media Player Keys and the following libraries: 
   libglib-2.0.so.0,
   libgobject-2.0.so.0,
   libgio-2.0.so.0

I don't know if Plasma 5 integrates with DBus but if it does then you should be able to get this add-on working by configuring media player keys and ensuring that the specified libs are installed.

**Comment by christophschw** [comment]
Thanks for your answer!

Unfortunately it doesn't work. As I said: In Gnome it's working flawless but not in Plasma... Maybe I should wait for the MPRIS integration. Would be the nicest solution :+1:

**Comment by carlin-q-scott** [comment]
Yeah, I would recommend waiting for MPRIS support if it's on Plasma's roadmap.

I don't plan to work on adding support for other DE's but other people can feel free to expand the support by forking the linuxMediaKeys.js file. That would require learning c-types (shouldn't take more than an afternoon) and then learning whatever API is provided by the DE for capturing key events.

**Comment by Noitidart** [comment]
XCB works in Plasma 5. I noted it to Carlin.

**Comment by gmelikov** [comment]
Please, look at #83 pull request, there are [instructions](https://github.com/gmelikov/browser-media-keys/blob/7feffdc17a50b428dfe04e828b6ae2869a9baebc/README.md) to simulate DBus services on other DE, it works on Xfce. I didn't use on KDE, but as i see here https://techbase.kde.org/Development/Tutorials/D-Bus/Introduction , KDE has DBus too.
